pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -185,20 +188,71 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 	} else if err != nil {
 		return err
 	} else {
-		r.log.Info("Updating StatefulSet",
-			"StatefulSet.Namespace", foundSts.Namespace,
-			"StatefulSet.Name", foundSts.Name)
-		zk.SyncStatefulSet(foundSts, sts)
-		err = r.client.Update(context.TODO(), foundSts)
-		if err != nil {
-			return err
+		foundSTSSize := *foundSts.Spec.Replicas
+		newSTSSize := *sts.Spec.Replicas
+		if newSTSSize < foundSTSSize {
+			// We're dealing with STS scale down
+			if !r.isConfigMapInSync(instance) {
+				r.log.Info("Skipping StatefulSet reconcile as ConfigMap not updated yet.")
+				return nil
+			}
+			//https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
+			r.skipSTSReconcile++
+			if r.skipSTSReconcile < 6 {
+				r.log.Info("Waiting for Config Map update to sync...Skipping STS Reconcile")
+				return nil
+			}
 		}
-		instance.Status.Replicas = foundSts.Status.Replicas
-		instance.Status.ReadyReplicas = foundSts.Status.ReadyReplicas
+		return r.updateStatefulSet(instance, foundSts, sts)
+	}
+	return nil
+}
+
+func (r *ReconcileZookeeperCluster) updateStatefulSet(instance *zookeeperv1beta1.ZookeeperCluster, foundSts *appsv1.StatefulSet, sts *appsv1.StatefulSet) (err error) {
+	r.log.Info("Updating StatefulSet",
+		"StatefulSet.Namespace", foundSts.Namespace,
+		"StatefulSet.Name", foundSts.Name)
+	zk.SyncStatefulSet(foundSts, sts)
+	err = r.client.Update(context.TODO(), foundSts)
+	if err != nil {
+		return err
 	}
+	instance.Status.Replicas = foundSts.Status.Replicas
+	instance.Status.ReadyReplicas = foundSts.Status.ReadyReplicas
+	r.skipSTSReconcile = 0
 	return nil
 }
 
+func (r *ReconcileZookeeperCluster) isConfigMapInSync(instance *zookeeperv1beta1.ZookeeperCluster) bool {
+	cm := zk.MakeConfigMap(instance)
+	foundCm := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      cm.Name,
+		Namespace: cm.Namespace,
+	}, foundCm)
+	if err != nil {
+		r.log.Error(err, "Error getting config map.")
+		return false
+	} else {
+		// found config map, now check numer of replicas in configMap
+		envStr := foundCm.Data["env.sh"]
+		r.log.Info("---ENV.SH---", "String", envStr)
```

Comment:
Looks like this is here for debugging?  Should it be removed before merge?

Reply:
Removed debug statements.